src/Microsoft.Azure.SignalR.Common/Endpoints/AccessKeyManager.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.SignalR
+{
+    internal class AccessKeySynchronizer : IAccessKeySynchronizer, IDisposable
+    {
+        private readonly ConcurrentDictionary<ServiceEndpoint, object> _endpoints = new ConcurrentDictionary<ServiceEndpoint, object>(new ServiceEndpointsComparer());
+
+        private readonly TimerAwaitable _timer = new TimerAwaitable(TimeSpan.Zero, TimeSpan.FromMinutes(5));
+
+        private readonly IServerNameProvider _provider;
+
+        private readonly ILoggerFactory _factory;
+
+        public AccessKeySynchronizer(
+            IServerNameProvider nameProvider,
+            ILoggerFactory loggerFactory,
+            bool test = false
```

Comment:
Not good to mixed flag here. How about make TimeAwaitable interval configurable as parameter for test use to shorten the wait time? Use a single constructor with comment as test use only.